File: dotnet/Families/GameX.Xbox/Formats/StardewValley/xTile.cs
using GameX.Xbox.Formats.Xna;
using System;
using System.IO;

namespace GameX.Xbox.Formats.StardewValley.xTile;

public class Map {
}

/// <summary>Tile Reader.</summary>
[RType("xTile.Pipeline.TideReader"), RAssembly("xTile")]
public class TideReader : TypeReader<Map> {
    public override Map Read(ContentReader r, Map o) {
        var data = r.ReadL32Bytes();
        return default;
        //Map map = FormatManager.Instance.BinaryFormat.Load((Stream)new MemoryStream(r.ReadBytes(count)));
        //if (map != null) {
        //    map.assetPath = r.AssetName;
        //    map.FlattenTileSheetPaths();
        //}
        //return map;
    }
}

File: python/gamex/families/Xbox/formats/xtile.py
from __future__ import annotations
import os
from openstk.core.typex import *
from gamex.families.Xbox.formats.xna import TypeReader

class Map:
    pass

@RType('xTile.Pipeline.TideReader')
@RAssembly('xTile')
class TideReader(TypeReader[Map]):
    def __init__(self, t: type): super().__init__(Map)
    def read(self, r: ContentReader, o: Map) -> Map:
        data = r.readL32Bytes()
        return Map()